### DISCUSSION TITLE
Unable to use chai-http with chai 5

### DISCUSSION THREAD

**Discussion by TerroR75** [chaijs/chai-http#332] [Support]
I tried pretty much everything I believe, i'm trying to get it working on .mjs files but constantly getting either "request is not a function" or chai does not provide an export named 'default'.
```js
import { expect, use } from "chai";
import { request } from "chai-http";

const chai = chaiModule.use(chaiHttp);
```
```js
import * as chaiModule from "chai";
import chaiHttp from "chai-http";

const chai = chaiModule.use(chaiHttp);
```

```js
import chaiModule from "chai";
import chaiHttp from "chai-http";

const chai = chaiModule.use(chaiHttp);
```

Nothing works.

**Comment by 43081j** [comment]
As per the readme, your second code snippet is correct

What version are you on? 

A reproduction is needed really. If you can give us an example of this not working, we can try help (an actual repo or a gist or something we can try ourselves)

**Reply by TerroR75** [reply]
@43081j Thank you for quick reply, here is a code gist: https://gist.github.com/TerroR75/2b4066e4288a930d5d6686548f4c4ac7
Hopefully it's enough. The errors I'm getting are inside user.test.js file as a comments but I'm gonna paste them here as well:

1. [DeprecationWarning] ...node_modules\chai-http\ has a "main" field set to "./index", excluding the full filename and extension to the resolved file at "index.js" ...
2. Exception during run: TypeError: chai.request is not a function

chai-http version is 5.0.0
chai version is 5.1.1
node version is 20.14.0

**Reply by 43081j** [reply]
The first one is an issue, good catch! We should update the `main` field to include `.js`. If you like, you could create a PR for that and I'll review (otherwise I'll try get to it next week). 

For the second point, you can likely do this: 

```ts
import * as chaiModule from 'chai';
import chaiHttp from 'chai-http';

const chai = chaiModule.use(chaiHttp);

// chai.request should now exist
```

In future, we should probably change this so you just import `request` from chai-http directly

**Reply by TobiTenno** [reply]
PR created: https://github.com/chaijs/chai-http/pull/333
